docs/readmes/proposals/p010_vendor_neutral_dp.md

```diff
@@ -416,7 +456,250 @@ scope of work.
+  according to the number of eNBs of each type.
+
+* **Radio Controller (RC)**
+  
+  RC is responsible for the CBSD to SAS communication
+  abstracted from vendor specific protocols.
+  
+  RC on one hand is responsible for placing the requests coming from PCs in the queue
+  and on the other for segregating the responses coming from Configuration Controller
+  and sending them back to the appropriate PC.
+
+* **Configuration Controller (CC)**
+
+  Configuration Controller is responsible for polling requests periodically from the queue,
+  aggregating them into one request in format understandable by SAS,
+  posting them over to SAS
+  and once a response is received, passing it on to RC.
+
+* **Message Queue**
+
+  RabbitMQ instance for storage of requests received by the RC
+  allowing them to be polled by a consumer.
+  Requests for which acknowledgement is received upon polling
+  will be deleted from the queue,
+  while the ones without acknowledgement will be requeued.
+
+* **Database (optional, to be implemented at stage 3)**
+
+  Relational DB storing the state of the DP.
+
+  DB allows to keep track of grants and CBSDs configuration limiting redundant CBSD - SAS connections.
+
+## Component Interactions
+
+* **Protocol Controller**
+
+  PC will pass the requests it receives from CBSDs to the RC over HTTPS.
+  
+  PC will pass the responses that are coming from RC back to the respective CBSDs
+  over the protocol that they are able to understand and the PC is implementing.
+
+* **Radio Controller**
+
+  RC receives requests from the PC over HTTPs and puts them in
+  the queue over AMQP.
+
+  When the response for the request comes back from CC,
+  RC will forward it over HTTPS to the appropriate PC.
+
+* **Configuration Controller**
+
+  CC communicates with the queue over AMQP for request polling.
+  
+  CC communicates with SAS (external) over HTTPS for posting bulked requests and with the RC over HTTPS for forwarding the SAS response.
+
+* **Message Queue**
+
+  Treating RC as the publisher and CC as the consumer.
+
+* **Database**
+
+  Exchanging information about CBSDs and grants with the CC.
+
+## Solution scaling
+
+The Domain Proxy will be split into separate services in separate containers.
+
+Every service will work in HA mode to provide service even in failover scenarios.
+
+Components such as database and queue will use cluster solutions
+delivered by database and queue providers.
+
+All other components will be stateless Python based services
+which do not need synchronization between them
+so it is not vital to create sync mechanisms.
+Python scripts will take over processing of data
+when one of the containers fail.
+When a container that processes requests received from RabbitMQ fails
+then RabbitMQ will not receive confirmation of received requests from the queue
+and will queue requests again.
+The only state will be kept in the database.
+
+## Batching of messages
```

Comment:
Can you elaborate on how messages are batched ? It would be great if some data model of request is also added to this document.

Reply:
All messages for the WInnForum protocol are defined in reference 1 TS-0016. Their API endpoints are message specific (Registration, SpectrumInquiry, Grants, Relinquishment etc.). But they can be batched, so that in bigger deployments, messages can be batched byte type/endpoint and sent as arrays. Not sure we need to duplicate the reference here.